x/sp/keeper/msg_server.go

```diff
@@ -4,8 +4,10 @@ import (
 	"context"
 
 	"cosmossdk.io/errors"
+	errorsmod "cosmossdk.io/errors"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	gov "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
```

Comment:
Why refer to `github.com/cosmos/cosmos-sdk/x/gov/types` here?
It is better to create own types here? 
Same for other modules for `UpdateParams`.